sfcli/SmartFace.Cli/Core/Domain/WatchlistMember/Impl/WatchlistMemberRegistrationManager.cs

```diff
@@ -47,12 +47,11 @@ public void RegisterWatchlistMember(RegisterWatchlistMemberExtended registerWatc
                 MIME = pathToPhotoFile.ToLower().EndsWith($".{Constants.PNG}") ? Constants.PNG_MIME_TYPE : Constants.JPEG_MIME_TYPE
             }));
 
-            Repository.Register(data);
-
-            Log.LogInformation($"WatchlistMember registered. [{data.ExternalId}]");
+            return Repository.RegisterAsync(data).ContinueWith(_ => Log.LogInformation($"WatchlistMember registered. [{data.ExternalId}]"));
```

Comment:
should be used with OnlyOnRanToCompletion option otherwise log will be written even in case of fault

Reply:
or not use ContinueWith at all here because exception handling is somehow specific when using it

Reply:
removed the continue with, method is now async